training/distributed_training/pytorch/model_parallel_v2/llama_v2/smp-finetuning-llama-fsdp-tp.ipynb

```diff
@@ -0,0 +1,959 @@
+   "source": [
+    "### Specify a Base Job Name"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "machine_str = instance_type.split(\".\")[1] + instance_type.split(\".\")[2][:3]\n",
+    "base_job_name = (\n",
+    "    f'smp-{model_config}-{machine_str}-hs{hybrid_shard_degree}-ao{offload_activations}-bs{hyperparameters[\"train_batch_size\"]}'\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "if not use_fsx:\n",
+    "    # If you want to resume training, set checkpoint_s3_uri to the same path as a previous job.\n",
+    "    # Previous checkpoint to load must have same model config.\n",
+    "    checkpoint_bucket = f\"s3://sagemaker-{region}-{account}/\"\n",
+    "    checkpoint_s3_uri = (\n",
+    "        f\"{checkpoint_bucket}/experiments/smp_fsdp-{model_type}-checkpoints/{base_job_name}/\"\n",
+    "    )"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Use the private preview image from ECR which contains all\n",
+    "# libraries and dependencies required for our SageMaker job.\n",
+    "image = CONTAINER_IMAGE"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "kwargs = {}\n",
+    "if use_fsx:\n",
+    "    # Use the security group and subnet that was used to create the fsx filesystem\n",
+    "    kwargs[\"security_group_ids\"] = [fsx_security_group_id]\n",
+    "    kwargs[\"subnets\"] = [fsx_subnet]\n",
+    "\n",
+    "smp_estimator = PyTorch(\n",
+    "    entry_point=\"train.py\",\n",
+    "    hyperparameters=hyperparameters,\n",
+    "    source_dir=os.getcwd(),\n",
+    "    role=role,\n",
+    "    checkpoint_s3_uri=checkpoint_s3_uri if not use_fsx else None,\n",
+    "    checkpoint_local_path=hyperparameters[\"checkpoint_dir\"] if use_fsx else None,\n",
+    "    instance_type=instance_type,\n",
+    "    volume_size=400,\n",
+    "    instance_count=instance_count,\n",
+    "    sagemaker_session=sagemaker_session,\n",
+    "    distribution={\n",
+    "        \"torch_distributed\": { \"enabled\": True }, # Use torchrun.\n",
+    "        \"smdistributed\": {\n",
+    "            \"modelparallel\": {\n",
+    "                \"enabled\": True,\n",
+    "                \"parameters\": {\n",
+    "                    \"tensor_parallel_degree\": tensor_parallel_degree,\n",
+    "                    \"hybrid_shard_degree\": hybrid_shard_degree,\n",
+    "                    \"sm_activation_offloading\": offload_activations,\n",
+    "                    \"activation_loading_horizon\": activation_loading_horizon,\n",
+    "                },\n",
+    "            }\n",
+    "        },\n",
+    "    },\n",
+    "    py_version=\"py310\",\n",
+    "    image_uri=image,\n",
```

Comment:
Please update to use `framework_version` instead

Reply:
Updated